### DISCUSSION TITLE
LSP ?

### DISCUSSION THREAD

**Discussion by Hetosk1** [helix-editor/helix#6082] [Q&A] [answered]
### Do LSP even work on windows ?
Well Hello guys i have been using helix/vim/nvim for more than a year and i've got my hands set on this CLI based text-editor, but I have a doubt............!
What is LSP and how to get my text editors integrated with it, I spent hours on youtube guys speaking on LSP but can't figure it out how to get started with it. I just know that it can help with some auto-complete suggestions but now i'm desperate to use this thing in my helix config. 
I'm using helix 22.12 on a windows terminal 
Please do help me with how to install a LSP of a programming language like c, rust, c++, java etc. Becuase those are the languages i code in.
And then please guide me on how to use that language server with a text editor like helix. And if this is not possible on windows please do inform me 
Thankyou for reading this article and waiting for a kind reply

**Comment by Proful** [comment]
Can you type ```hx --health rust``` & paste the output?

**Comment by kirawi** [comment, marked as answer]
https://github.com/helix-editor/helix/wiki/How-to-install-the-default-language-servers

**Comment by elazar125** [comment]
I got it working - I tried installing with chocolatey and had the same issue, I think that package is just broken. Compiling from source works though, and gives you color themes that you're probably missing too!
